peer/test_utils.go

```diff
@@ -643,3 +529,200 @@ func (m *mockMessageConn) LocalAddr() net.Addr {
 func (m *mockMessageConn) Close() error {
 	return nil
 }
+
+type peerTestCtx struct {
```

Comment:
On a second look, I think maybe it can be one struct. I have consolidated them.

Reply:
It looks like there is a commit in the middle where you have two copies of the same struct defined. You'll need to consolidate them in that fourth commit, rather than the last one, if you want the CI to accept it.